### PR TITLE
ansible-scylla-node: Update gpg keys before running apt update

### PR DESCRIPTION
**Pull request by igorribeiroduarte**
The task installing gnupg2 runs the equivalent to `apt update` before moving forward
due to `update_cache` being set to `true`. However if this role is being executed for
an old cluster which has an expired key, the `apt update` will fail.
In order to prevent that, this patch does the following:
  * Removes keyring file only if gnupg2 is already installed
  * Removes scylla repo keys only if gnupg2 is already installed
  * Installs gnupg2 only if it's not installed before setting up the keys

### CODE REVIEW

**Comment by vladzcloudius** [comment]
> The task installing gnupg2 runs the equivalent to `apt update` before moving forward due to `update_cache` being set to `true`. However if this role is being executed for an old cluster which has an expired key, the `apt update` will fail. In order to prevent that, this patch does the following:
> 
>     * Removes keyring file only if gnupg2 is already installed
> 
>     * Removes scylla repo keys only if gnupg2 is already installed
> 
>     * Installs gnupg2 only after removing the keys and file mentioned above and only if it's not already installed

This makes no sense: on the one hand you say that you can only remove keys only if `gnupg2` is already installed, but then you want to install `gnupg2` only after keys were removed but only if it isn't already installed. But it can't be not installed if you have removed the keys.

Bottom line: if you were able to remove keys this means that `gnupg2` is definitely installed - hence no point in installing it in this case.

> 
>     * Adds a task to update gnupg2 at the end of the process

**Review by vladzcloudius — CHANGES_REQUESTED** [review]
Same logical problem in the patch description.

File: ansible-scylla-node/tasks/Debian.yml [hunk 1]
```diff
@@ -61,6 +61,13 @@
     with_items: "{{ scylla_deb_repos }}"
     when: install_type == 'online' and item.split(".")[-1] == "list"
 
+  - name: Update gnupg2 dependency if necessary
```
Comment (vladzcloudius) [inline ansible-scylla-node/tasks/Debian.yml:64]: Why is this needed?

**Review by vladzcloudius — CHANGES_REQUESTED** [review]

File: ansible-scylla-node/tasks/Debian.yml (hunk 1 above)
Comment (igorribeiroduarte) [reply ansible-scylla-node/tasks/Debian.yml:64]: Without this `gnupg2` will never be updated by the role, since now the previous task installing `gnupg2` will only be executed if it's not already installed.

Comment (vladzcloudius) [reply ansible-scylla-node/tasks/Debian.yml:64]: Sure and this is how it was so far. Why is this a problem?

Comment (igorribeiroduarte) [reply ansible-scylla-node/tasks/Debian.yml]: Fixed

**Comment by igorribeiroduarte** [comment]
> > The task installing gnupg2 runs the equivalent to `apt update` before moving forward due to `update_cache` being set to `true`. However if this role is being executed for an old cluster which has an expired key, the `apt update` will fail. In order to prevent that, this patch does the following:
> > ```
> > * Removes keyring file only if gnupg2 is already installed
> > 
> > * Removes scylla repo keys only if gnupg2 is already installed
> > 
> > * Installs gnupg2 only after removing the keys and file mentioned above and only if it's not already installed
> > ```
> 
> This makes no sense: on the one hand you say that you can only remove keys only if `gnupg2` is already installed, but then you want to install `gnupg2` only after keys were removed but only if it isn't already installed. But it can't be not installed if you have removed the keys.
> 
> Bottom line: if you were able to remove keys this means that `gnupg2` is definitely installed - hence no point in installing it in this case.
> 
> > ```
> > * Adds a task to update gnupg2 at the end of the process
> > ```

Fixed

**Review by vladzcloudius — APPROVED** [review]
